src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixAbstractMethodsStep.cs

```diff
@@ -19,43 +19,80 @@ namespace MonoDroid.Tuner
 	/// <summary>
 	/// NOTE: this step is subclassed so it can be called directly from Xamarin.Android.Build.Tasks
 	/// </summary>
-	public class FixAbstractMethodsStep : BaseStep
+	public class FixAbstractMethodsStep
+#if NET5_LINKER
+	: IMarkHandler
+	{
+		protected LinkContext Context { get; private set; }
+		protected AnnotationStore Annotations => Context?.Annotations;
+
+		public void Initialize (LinkContext context, MarkContext markContext)
+		{
+			Context = context;
+			markContext.RegisterMarkTypeAction (type => ProcessType (type));
+		}
+#else
+	: BaseStep
 	{
 		readonly TypeDefinitionCache cache;
 
 		public FixAbstractMethodsStep (TypeDefinitionCache cache)
 		{
 			this.cache = cache;
 		}
+#endif
 
-		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		bool CheckShouldProcessAssembly (AssemblyDefinition assembly)
 		{
 			if (!Annotations.HasAction (assembly))
 				Annotations.SetAction (assembly, AssemblyAction.Skip);
 
 			if (IsProductOrSdkAssembly (assembly))
-				return;
+				return false;
 
 #if !NET5_LINKER
 			CheckAppDomainUsageUnconditional (assembly, (string msg) => Context.LogMessage (MessageImportance.High, msg));
 #endif
 
+			return assembly.MainModule.HasTypeReference ("Java.Lang.Object");
+		}
+
+		void UpdateAssemblyAction (AssemblyDefinition assembly)
+		{
+			AssemblyAction action = Annotations.HasAction (assembly) ? Annotations.GetAction (assembly) : AssemblyAction.Skip;
+			if (action == AssemblyAction.Skip || action == AssemblyAction.Copy || action == AssemblyAction.Delete)
+				Annotations.SetAction (assembly, AssemblyAction.Save);
```

Comment:
Do we really want to do this? I don't think we should change copy action to save not alone skip action to save.

Reply:
I don't have full context on how we can get into this situation - it looks to me like it was intended to inject methods into non-SDK assemblies which may be copy. I could get rid of skip/delete, but I'm trying to minimize functional changes here.

Reply:
It'd be fine to fix this in follow-up PR but I think this needs to be addressed because we changed the meaning of "copy" action in net6. It also does not make much sense to be to update code in deleted assembly.

Reply:
I got rid of the skip/delete cases because those don't make any sense to me. If anyone can provide more context on the copy case I'd be happy to address it in a follow-up.